libbeat/docs/upgrading.asciidoc

```diff
@@ -1,160 +1,329 @@
-changes>> doc.
+Before upgrading your {beats}, review the <<breaking-changes,breaking changes>>.
 
-// TODO: link to the stack upgrade docs
-//
-// If you are planning an upgrade of the full stack (Elasticsearch, Kibana, or
-// Logstash, in addition to Beats), please read the stack upgrade guide.
+If you're upgrading other products in the stack, also read the
+{stack-ref}/index.html[Elastic Stack Installation and Upgrade Guide]. 
+
+We recommend that you fully upgrade {es} and {kib} to version 7.0
+before upgrading {beats}. If you're on {beats} 6.0 through 6.6,
+upgrade to {beats} 6.7 *before* doing the {es} upgrade.
 
-We recommend that you fully upgrade Elasticsearch and Kibana to version 6.0
-before upgrading Beats. If you are on a Beats 5.x version lower than 5.6, please
-follow the <<upgrading-to-5.6>> section *before* doing the Elasticsearch
-upgrade.
+Upgrading between non-consecutive major versions (e.g. 5.x to 7.x) is not
+supported.
 
-If you're not ready to upgrade Elasticsearch and Kibana to 6.0, that's alright.
-Beats version 6.0 works with Elasticsearch and Kibana version 5.6, so you can
-upgrade Beats now and the rest of the stack later.
+IMPORTANT: Please read through all upgrade steps before proceeding. These steps
+are required before running the software for the first time.
 
-[[upgrading-to-5.6]]
-==== Upgrade to 5.6
+// TODO: We should add step-by-step instructions to tell users to back up the
+// registry, copy over the config file, etc. Similar to what Kibana does:
+// https://www.elastic.co/guide/en/kibana/master/upgrade-standard.html
 
-The upgrade procedure assumes that you have Beats version 5.6 installed. If you
-are on a previous 5.x version of Beats, please upgrade to version 5.6 first.
-The reason is that the Elasticsearch mapping template in 5.6 was modified to be
-compatible with Elasticsearch 6.0 (by removing the `_all` settings).
+[[upgrading-to-6.7]]
+==== Upgrade to {beats} 6.7 before upgrading to 7.0
 
-For this reason, after upgrading to 5.6, you need to make sure that the 5.6
-template is loaded. You can do this by temporarily enabling the
-`output.elasticsearch.template.overwrite` setting, for example with Metricbeat:
+The upgrade procedure assumes that you have {beats} 6.7 installed. If you're on
+a previous 6.x version of {beats}, upgrade to version 6.7 first. This is
+required because the {es} mapping template in 6.7 was modified to be compatible
+with {es} 7.0 (the `_type` setting changed from `doc` to `_doc`).
+
+After upgrading to 6.7, make sure the 6.7 template gets loaded. To load the
+template, run the `setup` command with the `--template` option specified. For
+example:
 
 [source,shell]
-------------------------------------------------------------------------------
-metricbeat -e -E output.elasticsearch.template.overwrite=true
-------------------------------------------------------------------------------
+----
+metricbeat setup --template
```

Comment:
For users using the LS output, this will fail. Should we add a note that for this to work, the ES output must be enabled?